pkg/ethereum/node_notifiee.go

```diff
@@ -127,6 +112,7 @@ func (n *Node) validatePeer(ctx context.Context, pid peer.ID, addrInfo peer.Addr
 		return false
 	}
 
+	n.sendMetadataEvent(ctx, pid, addrInfo, md)
```

Comment:
To elaborate on the mutext issue, I observe in the logs -
- 80 `Published metadata_received event` log which is from the `sendMetadataEvent`
- 5 `Performed successful handshake` log
which makes me think the issue is in `addToMetadataCache`. If I remove the mutex in `removeFromBackoffCache` this issue is solved